### ISSUE TITLE
Repository index missing or corrupted (Error 524)

### ISSUE THREAD

**Issue by irfanhakim-as** [borgbase/vorta#1708]
### Description

I booted into my PC, and Vorta has been set to autostart previously. I opened up Vorta to "compact" my backups, as yesterday before shutting down I had deleted a few backups. I did not get any errors using Vorta and my repository before today. As soon as I had opened up Vorta, I saw that the bottom of the Vorta window (next to the "Start Backup" button) was filled with a traceback of some error which I could not read since it was very long and cut off. I proceeded to try and do a "Compact" anyway, and it resulted to another long string of error traceback. I then tried creating a backup manually on Vorta, and got the same result. I was prompted to check the logs and so I did. The log is very long, so I will try and only include the snippets which contains warnings and errors.

### First snippet

This one seems like it might be unrelated to this issue:

```log
2023-05-06 05:06:33,107 - vorta.scheduler - DEBUG - Scheduling next run for 2023-05-06 06:01:07.869781
2023-05-06 05:07:30,101 - vorta.borg.borg_job - DEBUG - Cancel job on site default
2023-05-06 05:07:30,101 - vorta.borg.borg_job - DEBUG - Cancel job on site 1
2023-05-06 05:07:30,101 - vorta.borg.jobs_manager - INFO - Finished cancelling all jobs
2023-05-06 10:44:07,030 - vorta.i18n - DEBUG - Loading translation succeeded for ['en', 'en-US', 'en-Latn-US'].
2023-05-06 10:44:07,032 - vorta.scheduler - WARNING - Failed to connect to DBUS interface to detect sleep/resume events
2023-05-06 10:44:07,112 - root - DEBUG - Not a private SSH key file: authorized_keys
2023-05-06 10:44:07,134 - vorta.views.source_tab - DEBUG - Added item number 0 from 1
2023-05-06 10:44:07,222 - root - INFO - Using NetworkManagerMonitor NetworkStatusMonitor implementation.
```

### Second snippet

This error seems relevant, and likely came from when I had first booted up and it was trying to create a backup automatically. This includes a warning saying that my repository index is missing or corrupted. Checking this repository of mine, I did not find a file named `index`, whereas another repository of mine with backups for another PC does have one:

```log
2023-05-06 10:44:08,444 - vorta.keyring.abc - DEBUG - Using VortaKWallet5Keyring
2023-05-06 10:44:08,444 - vorta.borg.borg_job - DEBUG - Using VortaKWallet5Keyring keyring to store passwords.
2023-05-06 10:44:08,446 - vorta.keyring.kwallet - DEBUG - Retrieved password for repo /mnt/nas/Snapshots/calcifer
2023-05-06 10:44:08,452 - vorta.scheduler - INFO - Preparation for backup successful.
2023-05-06 10:44:08,453 - vorta.borg.jobs_manager - DEBUG - Add job for site 1
2023-05-06 10:44:08,453 - vorta.borg.jobs_manager - DEBUG - Start job on site: 1
2023-05-06 10:44:08,455 - vorta.borg.borg_job - INFO - Running command /app/bin/borg create --list --progress --info --log-json --json --filter=AM -C lz4 --exclude-from /tmp/tmpnijvxkhd --exclude-if-present .nobackup /mnt/nas/Snapshots/calcifer::calcifer-2023-05-06-104408 /home/foobar
2023-05-06 10:44:09,686 - vorta.borg.borg_job - WARNING - Repository index missing or corrupted, trying to recover from: [Errno 524] Unknown error 524
2023-05-06 10:44:13,850 - vorta.borg.borg_job - ERROR - Local Exception
2023-05-06 10:44:13,851 - vorta.borg.borg_job - ERROR - Traceback (most recent call last):
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 526, in open_index
    return NSIndex.read(fd)
  File "src/borg/hashindex.pyx", line 114, in borg.hashindex.IndexBase.read
  File "src/borg/hashindex.pyx", line 99, in borg.hashindex.IndexBase.__cinit__
  File "/app/lib/python3.10/site-packages/borg/crypto/file_integrity.py", line 32, in read
    return self.fd.read(n)
  File "/app/lib/python3.10/site-packages/borg/crypto/file_integrity.py", line 82, in read
    data = super().read(n)
  File "/app/lib/python3.10/site-packages/borg/crypto/file_integrity.py", line 32, in read
    return self.fd.read(n)
OSError: [Errno 524] Unknown error 524

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 5213, in main
    exit_code = archiver.run(args)
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 5144, in run
    return set_ec(func(args))
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 170, in wrapper
    kwargs['manifest'], kwargs['key'] = Manifest.load(repository, compatibility)
  File "/app/lib/python3.10/site-packages/borg/helpers/manifest.py", line 183, in load
    cdata = repository.get(cls.MANIFEST_ID)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1201, in get
    self.index = self.open_index(self.get_transaction_id())
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 532, in open_index
    self.prepare_txn(self.get_transaction_id())
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 412, in get_transaction_id
    self.check_transaction()
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 409, in check_transaction
    self.replay_segments(replay_from, segments_transaction_id)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 902, in replay_segments
    self._update_index(segment, objects)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 912, in _update_index
    for tag, key, offset, size in objects:
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1512, in iter_objects
    size, tag, key, data = self._read(fd, self.header_fmt, header, segment, offset,
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1606, in _read
    data = fd.read(length)
OSError: [Errno 524] Unknown error 524

Platform: Linux calcifer 6.3.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Mon, 01 May 2023 17:42:39 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.2.4  Python: CPython 3.10.6 msgpack: 1.0.4 fuse: pyfuse3 3.2.2 [pyfuse3,llfuse]
PID: 25  CWD: /home/foobar
sys.argv: ['/app/bin/borg', 'create', '--list', '--progress', '--info', '--log-json', '--json', '--filter=AM', '-C', 'lz4', '--exclude-from', '/tmp/tmpnijvxkhd', '--exclude-if-present', '.nobackup', '/mnt/nas/Snapshots/calcifer::calcifer-2023-05-06-104408', '/home/foobar']
SSH_ORIGINAL_COMMAND: None
```

There were a bunch of other errors like this in the log, from attempts to create a backup.

### Third snippet

This error is also similar, but from my attempt to perform a "compact":

```log
2023-05-06 10:57:31,105 - vorta.borg.borg_job - INFO - Running command /app/bin/borg --info --log-json compact --progress /mnt/nas/Snapshots/calcifer
2023-05-06 10:57:36,409 - vorta.borg.borg_job - ERROR - Local Exception
2023-05-06 10:57:36,409 - vorta.borg.borg_job - ERROR - Traceback (most recent call last):
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 5213, in main
    exit_code = archiver.run(args)
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 5144, in run
    return set_ec(func(args))
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 183, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 1852, in do_compact
    data = repository.get(Manifest.MANIFEST_ID)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1201, in get
    self.index = self.open_index(self.get_transaction_id())
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 412, in get_transaction_id
    self.check_transaction()
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 409, in check_transaction
    self.replay_segments(replay_from, segments_transaction_id)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 902, in replay_segments
    self._update_index(segment, objects)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 912, in _update_index
    for tag, key, offset, size in objects:
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1512, in iter_objects
    size, tag, key, data = self._read(fd, self.header_fmt, header, segment, offset,
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1606, in _read
    data = fd.read(length)
OSError: [Errno 524] Unknown error 524

Platform: Linux calcifer 6.3.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Mon, 01 May 2023 17:42:39 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.2.4  Python: CPython 3.10.6 msgpack: 1.0.4 fuse: pyfuse3 3.2.2 [pyfuse3,llfuse]
PID: 45  CWD: /home/foobar
sys.argv: ['/app/bin/borg', '--info', '--log-json', 'compact', '--progress', '/mnt/nas/Snapshots/calcifer']
SSH_ORIGINAL_COMMAND: None
```

Seems to me that my issue has everything to do with the `index` file that's somehow now been missing. May I know how this file may have ended up missing, and what are my options now?

### Reproduction

- [X] I tried to reproduce the issue.
- [X] I was able to reproduce the issue.

### OS

Linux 6.3.1-arch1-1, KDE Plasma 5.27.4

### Version of Vorta

0.8.12

### What did you install Vorta with?

Flatpak

### Version of Borg

1.2.4

### Logs

```pytb
2023-05-06 11:33:59,811 - vorta.scheduler - INFO - Preparation for backup successful.
2023-05-06 11:33:59,811 - vorta.borg.jobs_manager - DEBUG - Add job for site 1
2023-05-06 11:33:59,811 - vorta.borg.jobs_manager - DEBUG - Start job on site: 1
2023-05-06 11:33:59,813 - vorta.borg.borg_job - INFO - Running command /app/bin/borg create --list --progress --info --log-json --json --filter=AM -C lz4 --exclude-from /tmp/tmpnu3wqsge --exclude-if-present .nobackup /mnt/nas/Snapshots/calcifer::calcifer-2023-05-06-113359 /home/foobar
2023-05-06 11:34:00,288 - vorta.notifications - DEBUG - notification not suppressed
2023-05-06 11:34:00,290 - vorta.scheduler - ERROR - Error during backup creation.
2023-05-06 11:34:00,291 - vorta.scheduler - DEBUG - Paused 1 until 2023-05-06 12:04:00
2023-05-06 11:34:00,292 - vorta.scheduler - DEBUG - Nothing scheduled for profile 1 because of timeout until 2023-05-06 12:04:00.
2023-05-06 11:34:04,366 - vorta.borg.borg_job - ERROR - Local Exception
2023-05-06 11:34:04,367 - vorta.borg.borg_job - ERROR - Traceback (most recent call last):
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 5213, in main
    exit_code = archiver.run(args)
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 5144, in run
    return set_ec(func(args))
  File "/app/lib/python3.10/site-packages/borg/archiver.py", line 170, in wrapper
    kwargs['manifest'], kwargs['key'] = Manifest.load(repository, compatibility)
  File "/app/lib/python3.10/site-packages/borg/helpers/manifest.py", line 183, in load
    cdata = repository.get(cls.MANIFEST_ID)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1201, in get
    self.index = self.open_index(self.get_transaction_id())
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 412, in get_transaction_id
    self.check_transaction()
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 409, in check_transaction
    self.replay_segments(replay_from, segments_transaction_id)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 902, in replay_segments
    self._update_index(segment, objects)
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 912, in _update_index
    for tag, key, offset, size in objects:
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1512, in iter_objects
    size, tag, key, data = self._read(fd, self.header_fmt, header, segment, offset,
  File "/app/lib/python3.10/site-packages/borg/repository.py", line 1606, in _read
    data = fd.read(length)
OSError: [Errno 524] Unknown error 524

Platform: Linux calcifer 6.3.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Mon, 01 May 2023 17:42:39 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.2.4  Python: CPython 3.10.6 msgpack: 1.0.4 fuse: pyfuse3 3.2.2 [pyfuse3,llfuse]
PID: 26  CWD: /home/foobar
sys.argv: ['/app/bin/borg', 'create', '--list', '--progress', '--info', '--log-json', '--json', '--filter=AM', '-C', 'lz4', '--exclude-from', '/tmp/tmpnu3wqsge', '--exclude-if-present', '.nobackup', '/mnt/nas/Snapshots/calcifer::calcifer-2023-05-06-113359', '/home/foobar']
SSH_ORIGINAL_COMMAND: None

2023-05-06 11:34:04,486 - vorta.borg.jobs_manager - DEBUG - Finish job for site: 1
2023-05-06 11:34:04,487 - vorta.scheduler - DEBUG - Nothing scheduled for profile 1 because of timeout until 2023-05-06 12:04:00.
2023-05-06 11:34:04,487 - vorta.borg.jobs_manager - DEBUG - No more jobs for site: 1
2023-05-06 11:34:04,973 - vorta.notifications - DEBUG - notification not suppressed
2023-05-06 11:34:04,976 - vorta.scheduler - ERROR - Error during backup creation.
```

**Comment by m3nu** [comment]
> `OSError: [Errno 524] Unknown error 524`

Is your repo mounted on a network drive or FUSE?

**Comment by bitwave** [comment]
I have the same problem with borg only (no vorta). Repo is mounted as samba share.

**Comment by m3nu** [comment]
I guess that solves it then. Borg expects a local file system. Anything else can always cause issues, so use at your own risk. Or use SSH for the remote connection.

**Comment by bitwave** [comment]
The problem (kind of) is, that CIFS/SMB mounted shares worked before.

**Comment by irfanhakim-as** [comment]
> > `OSError: [Errno 524] Unknown error 524`
> 
> Is your repo mounted on a network drive or FUSE?

Both my repos are SMB shares mounted using the following line in `/etc/fstab`:

```
//nas.domain/nas   /mnt/nas   cifs    _netdev,nofail,mfsymlinks,credentials=/home/foobar/.smbcreds,vers=3.0,uid=1000,gid=1000,iocharset=utf8   0 0
```

They have worked flawlessly for almost a year now. Other backup solutions besides Vorta have been kind of iffy when it comes to using a network drive - so I'm absolutely thankful for Vorta/Borg.

The one notable thing I had in mind was that a day before this issue had occurred with *one* of my repo is that I had only just changed the hostname of my NAS server on the server itself and on my client machine using Vorta, but I think this probably is a non-issue since the sharepoint (`/mnt/nas`) is still the same (repo uses `/mnt/nas/Snapshots/calcifer`) and `/etc/fstab` has been updated accordingly, and the share was also mounted automatically perfectly fine.

Do you have a suggestion what I can do to prevent this (if any) or any insight on why this even happened in the first place since they've worked perfectly fine before, and most importantly what should I do now (eg. Do I simply have to delete the old repo with its backup data and create a new repo)?

Thank you.

**Comment by ArchieLinux** [comment]
Having a similar issue - Vorta worked like a charm for many months on this Arch Linux setup, as it continues to on other setups in the house. But in the last day, after deleting the latest back within Vorta (using Vorta's delete archive function, successfully) now I get a stream of traceback errors when trying to start up Vorta again.

I wound up uninstalling Vorta and Borg - and deleting my entire backup archive and starting from scratch... but those pesky traceback errors continue and I cannot get Vorta to start up...

```
Traceback (most recent call last):
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/vorta", line 33, in <module>
    sys.exit(load_entry_point('vorta==0.8.12', 'gui_scripts', 'vorta')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/bin/vorta", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 981, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for vorta
```

Any insights appreciated!

**Comment by irfanhakim-as** [comment]
> Having a similar issue - Vorta worked like a charm for many months on this Arch Linux setup, as it continues to on other setups in the house. But in the last day, after deleting the latest back within Vorta (using Vorta's delete archive function, successfully) now I get a stream of traceback errors when trying to start up Vorta again.

Huh, are you saying your issue started after, like me, successfully deleting some archives as the last thing you did on Vorta before next when you open it up for the first time again which is when the traceback errors started showing up? cos if yes, then that might be interesting and _maybe_ related.

I delete some of those minimal/"differential" archives pretty often with no issues in the past, just like this time. Archives I delete also always include either the latest one or latest two-three of them, and not random, cos it just seemed safer to me to do it that way. Except, normally I would "compact" them immediately afterwards, but I didn't this time because it was late and I had to turn off my PC already. not sure why that would cause the issue though, so idk.

**Comment by ArchieLinux** [comment]
@irfanhakim-as Yes, indeed, that's exactly what I'm saying.

I have vorta running beautifully on two other installations. All three have been kept current with latest releases of Arch using yay -Syu. The only difference is that on the one that's now unable to function I just deleted the most recent incremental backup created using vorta's delete function.

Now I cannot get vorta to run whatsoever. And I've deleted the entire backup drive, uninstalled both vorta and borg and reinstalled. No dice. Same errors as shown above through terminal and vorta won't start up.

:shrug:

**Comment by irfanhakim-as** [comment]
Since there was no recommendation on what to do, here's what I did:

- Unlinked the repository from Vorta
- Deleted the repo folder from my SMB mount and create a new one
- Create a new "local folder" repo on Vorta and point to the newly created folder
- Using the same profile I had used before, run "Start Backup" once again and start from scratch

I'd try using the remote repo but I intend to store my backup data on my NAS, and I don't think it'd be good for me to enable SSH on it all the time. Also from past experience with other backup solutions, backing up huge chunks to my local server/NAS via SSH bogged my network really hard which often led to a partially done backup and my network needing a restart. Using an SMB mount though works every time.

Everything has worked out fine so far. After this, I'd probably be more careful about deleting my archives before shutting down (eventho they were successful). @ArchieLinux I'd recommend using the Flatpak version of Vorta instead, it works just as you would expect from the AUR version.

**Comment by ArchieLinux** [comment]
> . @ArchieLinux I'd recommend using the Flatpak version of Vorta instead, it works just as you would expect from the AUR version.

Excellent thought. I've tried to avoid flatpacks on this particular installation, but you've tempted me. Meanwhile, here's what I tried instead and it SEEMS to be working ...

After uninstalling both vorta and borg (`yay -Rc vorta; yay -Rc borg`), I then rebooted and went back to yay for a fresh reinstall of vorta (`yay -S vorta`) which brings borg along with it as a dependency.

But this time, during the reinstallation process, I am careful to answer the yay app questions for a CLEAN install. (Normally, I never bother answering that step in such a way as to require a clean install, and just hit enter for all the questions. This was a key change in my procedure and I believe made the difference) ...

This time it took more time, reloading things from scratch. And then I rebooted again.

And it took some time for vorta to fire up - apparently just like the first time like I vaguely remember from long ago. But, it currently seems to be working perfectly.

So my suggestion would be try a "clean" install during the yay process. 

FWIW, I had already wiped out my history/backup directory. So I had to start vorta from ground zero and recreate my backup directory/location. But whether that was part of my solution is impossible for me to test at this stage.

If I encounter some weird behavior issues or additional errors again I will update this post accordingly. But the clean install has at least brought vorta back to life for me. Until now, I couldn't even get the darn thing to run. All looks normal so far.
 
Hope some of this might be helpful.

**Comment by irfanhakim-as** [comment]
@ArchieLinux I'm glad that had worked out for you. Installing it from the AUR might fix my issue, but I'm still going to explore other solutions without needing to replacing my Flatpak Vorta install.

Here's my update so far:

- I prefer the Flatpak version because theoretically, it should be *more* of a containerised/isolated solution, which means uninstalling it and deleting its data, then reinstalling it again should offer a "brand new" experience. So far, doesn't seem true as creating a new repo on the NAS and backing up, while successful, still leaves a bunch of `Errno 524` warnings.
- I know for a fact that it has nothing to do with the NAS and possibly the mount because my other machines use the exact same setup (besides their own repo/folder on the same NAS) and they still backup/work fine to this day.
- Vorta on those other machines were also installed using Flatpak, and are all of the same version of `vorta` and `borg`. The only notable thing I can think of is their `python` (potentially other dependencies) installed **in the system** may be different BUT that shouldn't be a problem for Flatpaks, no? My understanding is if their `com.borgbase.Vorta` versions are the same, they should behave the same.
- I suspected that it might have been caused by my `/etc/hosts` or `ssh` setup somehow. Since before this started occurring, I had changed the hostname of my NAS on my `/etc/hosts` file and also my `/etc/fstab`, from a not real domain, to an actual domain. I thought it maybe somehow affected with how Vorta sees my `/mnt/nas` mountpoint, and that it might had tried to contact the actual domain. So I tried reverting my `/etc/hosts` file and my `/etc/fstab`, which initially seemed like it might had worked, but when I experimented with those files again and the error had returned, reverting them back the second time (with restarts upon every attempts etc) did not seem to have fixed it.
- With all this in mind, I'm almost certain that a complete reinstall would make my Vorta work again as it did before for the past year, and as it still does on my other machines. But a complete reinstall is definitely not an option now as this is my main system.

I'm at my limit now on figuring out how to completely reset this Vorta install for a fresh, new experience which I hope would fix this mess - unless @m3nu or someone else can come forward with any tip I can explore. A local/external drive repo is not an option at the moment with my limited desk space and limited I/O on my SFF PC, and frankly, I really don't see a reason why I could not continue using my NAS through a local mount as the repo when it has worked before, and it still works on my other devices.

**Comment by real-yfprojects** [comment]
I just saw this long thread. Since this isn't a Vorta but a borg issue, maybe ask Thomas. What do you think @m3nu?

**Comment by ArchieLinux** [comment]
> I just saw this long thread. Since this isn't a Vorta but a borg issue, maybe ask Thomas. What do you think @m3nu?

I'm somewhat agnostic on where the topic belongs, since users of Vorta are also users of Borg. But my impression is that my issue revolved around how I could no longer run Vorta (it got mis-installed) after I deleted an archive using Vorta's GUI functionality. 

For my situation, the solution was to do a clean install of Vorta. To me, not having accessed Borg directly, this suggests mine was a Vorta issue. :shrug:

**Comment by bitwave** [comment]
I have the issue still open for borg only. Is there a related tracking issue for borg already open?

**Comment by CrashTD** [comment]
It looks like this is a python problem, not vorta or borg. i had the same on Manjaro ` File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 563,` etc. and stumbled over a reddit post linking to this: https://forum.manjaro.org/t/unstable-update-2023-05-03-python-3-11-gcc-13/139877

> You will need to rebuild any AUR Python packages that install files to site-packages. :information_source:
> 
> Print a list of them:
> 
> pacman -Qoq /usr/lib/python3.10
> Rebuild them all at once:*
> 
> pamac build $(pacman -Qoq /usr/lib/python3.10)

rebuilding vorta-git from the AUR worked for me.

**Comment by irfanhakim-as** [comment]
> It looks like this is a python problem, not vorta or borg. i had the same on Manjaro ` File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 563,` etc. and stumbled over a reddit post linking to this: https://forum.manjaro.org/t/unstable-update-2023-05-03-python-3-11-gcc-13/139877
> 
> > You will need to rebuild any AUR Python packages that install files to site-packages. information_source
> > Print a list of them:
> > pacman -Qoq /usr/lib/python3.10
> > Rebuild them all at once:*
> > pamac build $(pacman -Qoq /usr/lib/python3.10)
> 
> rebuilding vorta-git from the AUR worked for me.

I actually somewhat suspected this, bcos I did do a system update recently prior to the issue. I have also noticed a bunch of `python` libraries missing or not working, that required me to reinstall them again. I'm not sure if my `python` was of version `3.10` before, but it's `3.11.3` now.

I just thought that Flatpak apps should be exempted from this issue, since I'm pretty sure, even from my digging, that they have their own `python` that they use that is version `3.10`. Will still try i guess when I have the time soon. Thanks!

**Comment by BeK3103** [comment]
It doesn't seem to me as if this is a Python issue.  Even when just running a command on the files on the CIFS drive from a terminal, I get err 524:
`cmp file1 file2cmp: file1: unknown error 524`

**Comment by ArchieLinux** [comment]
> It doesn't seem to me as if this is a Python issue. Even when just running a command on the files on the CIFS drive from a terminal, I get err 524:
> `cmp file1 file2cmp: file1: unknown error 524`

I've kind of lost track of the genesis of my experience with these error messages, but one those (now rare) occasions when I experience this phenomenon the solution is quick and easy each time ... No need to mess with the actual archives themselves (deleting one or more vs not deleting any of them makes no difference), the solution for me has simply been to reinstall Vorta (which includes Borg) as a CLEAN BUILD option when using `yay`. 

In other words, when you are responding to `yay` prompts during the installation process, make sure not to skip by the option to "clean build" the program - as this has been the effective solution in my situation.

Not sure if this has been a Python issue or not - but that seems to be the case based on the fleeting on-screen messages during the rebuild process. TBH, I haven't done a deeper dive than this since it fully resolves my error issues. 

HTH

**Comment by real-yfprojects** [comment]
> It doesn't seem to me as if this is a Python issue.  Even when just running a command on the files on the CIFS drive from a terminal, I get err 524:
> `cmp file1 file2cmp: file1: unknown error 524`

Since this is an OSError it is raised by the OS or the drivers of the device. I found this interesting comment on Error 524:
https://github.com/vpelletier/python-functionfs/issues/2#issuecomment-342326329

This suggests that the Error is issued by the driver in case of unsupported operations. So this is neither a Vorta nor a borg problem. That's why I will close this issue, you should still be able to exchange your experiences here.
